### PR TITLE
write FeeCollector once for each block

### PR DESCRIPTION
**Pull request by MinZhu09**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/cosmos-sdk/x/auth/ante/fee.go [hunk 1]
```diff
@@ -116,33 +116,39 @@ func (dfd DeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bo
 //
 // NOTE: We could use the BankKeeper (in addition to the AccountKeeper, because
 // the BankKeeper doesn't give us accounts), but it seems easier to do this.
-func DeductFees(supplyKeeper types.SupplyKeeper, ctx sdk.Context, acc exported.Account, fees sdk.Coins) error {
+func DeductFees(ak keeper.AccountKeeper, ctx sdk.Context, acc exported.Account, fees sdk.Coins) (sdk.Context, error) {
```
Comment (BananaLF) [inline libs/cosmos-sdk/x/auth/ante/fee.go]: this function may be trouble, because gas calculate is different form old code. If cosmos_tx may be cause smb.

**Review by xzav3r — DISMISSED** [review]

File: libs/cosmos-sdk/x/auth/types/account.go [hunk 2]
```diff
@@ -244,6 +246,9 @@ func (acc *BaseAccount) GetCoins() sdk.Coins {
 
 // SetCoins - Implements sdk.Account.
 func (acc *BaseAccount) SetCoins(coins sdk.Coins) error {
+	if hex.EncodeToString(acc.GetAddress()) == "c82854bbd93e996e7d279f5038dd70e71da7f026" {
```
Comment (BananaLF) [inline libs/cosmos-sdk/x/auth/types/account.go]: the line is useless code

File: x/evm/types/statedb.go [hunk 3]
```diff
@@ -872,6 +872,7 @@ func (csdb *CommitStateDB) updateStateObject(so *stateObject) error {
 	if err := so.account.SetCoins(coins); err != nil {
 		return err
 	}
+	//fmt.Println("updateState. ", hex.EncodeToString(so.account.GetAddress()))
```
Comment (BananaLF) [inline x/evm/types/statedb.go]: need remove this line

File: app/app_parallel.go [hunk 4]
```diff
@@ -13,17 +13,20 @@ import (
 // feeCollectorHandler set or get the value of feeCollectorAcc
 func updateFeeCollectorHandler(bk bank.Keeper, sk supply.Keeper) sdk.UpdateFeeCollectorAccHandler {
 	return func(ctx sdk.Context, balance sdk.Coins) error {
+		sk.GetModuleAccount(ctx, auth.FeeCollectorName)
```
Comment (BananaLF) [inline app/app_parallel.go]: this line is useless

File: libs/cosmos-sdk/baseapp/abci.go [hunk 5]
```diff
@@ -161,11 +161,29 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 
 	app.anteTracer = trace.NewTracer(trace.AnteChainDetail)
 
+	app.feeForCollector = sdk.Coins{}
+
 	return res
 }
 
+func (app *BaseApp) UpdateFeeForCollector(fee sdk.Coins, add bool) {
```
Comment (BananaLF) [inline libs/cosmos-sdk/baseapp/abci.go:170]: the time call this function must be deliver_tx or parallel_tx

File: libs/cosmos-sdk/baseapp/baseapp_parallel.go [hunk 6]
```diff
@@ -78,7 +78,7 @@ func (app *BaseApp) paraLoadSender(txs [][]byte) {
 			for txBytes := range ch {
 				tx, err := app.txDecoder(txBytes)
 				if err == nil {
-					app.getTxFee(checkStateCtx.WithTxBytes(txBytes), tx)
+					app.getTxFee(checkStateCtx.WithTxBytes(txBytes), tx, true)
```
Comment (BananaLF) [inline libs/cosmos-sdk/baseapp/baseapp_parallel.go]: this line may be set flase，i think this function is parallel verify signature.  so must be set fasle to verify signature.

**Review by BananaLF — DISMISSED** [review]

**Review by cwbhhjl — DISMISSED** [review]

**Review by BananaLF — APPROVED** [review]

**Pull request merged**